src/contract.rs

```diff
@@ -73,9 +76,120 @@ pub fn execute(
             validator_oper_addr,
         } => remove_validator(deps, info, validator_oper_addr),
         ExecuteMsg::WithdrawFunds { amount } => withdraw_funds(deps, info, amount),
+
```

Comment:
Can we add a check in record_metrics to ensure the recording timestamp is greater than or equal to the last timestamp in state?

Reply:
Okay adding this check. Shouldn't the recording timestamp also be greater than existing timestamps, rather than greater or equal?

Reply:
Aren't we talking about the same thing?

Reply:
greater than vs greater than or equal to is the difference. Have made it greater than in the code!